src/client/clientmap.h

```diff
@@ -86,10 +86,24 @@ class ClientMap : public Map, public scene::ISceneNode
 
 	void updateCamera(const v3f &pos, const v3f &dir, f32 fov, const v3s16 &offset)
 	{
+		v3s16 previous_node = floatToInt(m_camera_position, BS);
+		v3s16 previous_block = getContainerPos(previous_node, MAP_BLOCKSIZE);
+
 		m_camera_position = pos;
 		m_camera_direction = dir;
 		m_camera_fov = fov;
 		m_camera_offset = offset;
+
+		v3s16 current_node = floatToInt(m_camera_position, BS);
+		v3s16 current_block = getContainerPos(current_node, MAP_BLOCKSIZE);
+
+		// if moved over node boundary
+		if (current_node != previous_node)
+			markBlocksDirty(current_node, previous_node, current_block, previous_block);
+
+		// reorder the blocks when camera crosses block boundary
+		if (previous_block != current_block)
+			updateDrawList();
```

Comment:
updateDrawList is called roughly 5 times a second, see [game.cpp#3833](
https://github.com/minetest/minetest/blob/a24899bf2dcd58916922d671ee8761448b6876e5/src/client/game.cpp#L3833).

I need to force-update draw list when a block boundary is crossed to reorder the blocks by new Manhattan distances and avoid flicker. It may be possible to change the condition in game.cpp, but I found this call easier.

Reply:
I've now replaced this call with a flag for Game to call updateDrawList from the game loop.